core/src/main/scala/org/http4s/blaze/pipeline/stages/SSLStage.scala

```diff
@@ -277,33 +281,36 @@ final class SSLStage(engine: SSLEngine, maxWrite: Int = 1024*1024) extends MidSt
   }
 
   // Attempts to continue write requests
-  private def continueWrite(data: Array[ByteBuffer], p: Promise[Unit]): Unit = handshakeQueue.synchronized {
-    if (inHandshake()) handshakeQueue += DelayedWrite(data, p)
-    else {
-      val out = new ArrayBuffer[ByteBuffer]
-      writeLoop(data, out) match {
-        case SSLSuccess if BufferTools.checkEmpty(data) =>
-          p.completeWith(channelWrite(out))
-
-        case SSLSuccess => // must have more data to write
-          channelWrite(out).onComplete {
-            case Success(_) => continueWrite(data, p)
-            case Failure(t) => p.tryFailure(t)
-          }(trampoline)
-
-        case SSLNeedHandshake(r) =>
-          handshakeQueue += DelayedWrite(data, p)
-          val readData = takeQueuedBytes()
-          if (out.nonEmpty) { // need to write
+  private def continueWrite(data: Array[ByteBuffer], p: Promise[Unit]): Unit = {
+    handshakeQueue.synchronized {
+      if (inHandshake()) handshakeQueue += DelayedWrite(data, p)
+      else {
+        val out = new ArrayBuffer[ByteBuffer]
+        writeLoop(data, out) match {
+          case SSLSuccess if BufferTools.checkEmpty(data) =>
+            p.completeWith(channelWrite(out))
+
+          case SSLSuccess => // must have more data to write
             channelWrite(out).onComplete {
-              case Success(_) => sslHandshake(readData, r)
-              case Failure(t) => handshakeFailure(t)
+              case Success(_) => continueWrite(data, p)
```

Comment:
Is this a reversion that needs to be fixed, I was working with older data and the rebase may have overridden.